src/js/trove/particle.js

```
@@ -0,0 +1,154 @@
+                ret[k] = v;
+              } else {
+                throw new Error('unimplemented value conversion for StringDict');
+              }
+            }
+            return ret;
+          };
+
+          var OnParticle = function(handler, name, sdict) {
+            WorldConfigOption.call(this, 'on-particle');
+            this.handler = handler;
+            this.event = name;
+            this.options = sd_to_js(sdict);
+          };
+
+          OnParticle.prototype = Object.create(WorldConfigOption.prototype);
+
+          OnParticle.prototype.toRawHandler = function(toplevelNode) {
+            var that = this;
+            var worldFunction = adaptWorldFunction(that.handler);
+            return rawJsworld.on_particle(worldFunction, that.event, that.options);
+          };
+
+
+          //////////////////////////////////////////////////////////////////////
+          
+          var ToParticle = function(handler, ename, sdict) {
+            WorldConfigOption.call(this, 'to-particle');
+            this.handler = handler;
+            this.event = ename;
+            this.options = sd_to_js(sdict);
+          };
+          
+          ToParticle.prototype = Object.create(WorldConfigOption.prototype);
+          
+          ToParticle.prototype.toRawHandler = function(toplevelNode) {
+            var that = this;
+            var worldFunction = adaptWorldFunction(that.handler);
+            var eventGen = function(w, k) {
+              worldFunction(w, function(v) {
+                if(ffi.isSome(v)) {
+                  var xhr = new XMLHttpRequest();
+                  xhr.open("POST","https://api.particle.io/v1/devices/events");
+                  xhr.setRequestHeader("Content-type","application/x-www-form-urlencoded");
+                  xhr.send("access_token=" + that.options.acc +
+                           "&name=_event&data=" + that.event + ":" +
+                           runtime.getField(v, "value") + "&private=true&ttl=60");
```

What are the situations where this could fail?  It seems like a failure to POST will be silently ignored (which might be the intended semantics), but that could also be kind of baffling.